admin/tool/policy/classes/api.php

```diff
@@ -396,4 +402,118 @@ public static function move_up($policyid) {
     public static function move_down($policyid) {
         static::move_policy_document($policyid, 3);
     }
+
+    /**
+     * Returns list of acceptances
+     *
+     * @param int|array $users id of a user or a list of ids
+     * @param int|array $versions list of policy versions
+     * @return array list of acceptances indexed by userid and versionid. Each acceptance has additional field 'policyid'
+     */
+    public static function get_acceptances($versions, $select = '', $params = [], $orderby = 'u.lastname, u.firstname', $limitfrom = 0, $limitto = 0) {
+        global $DB;
+        $versions = is_array($versions) ? array_values($versions) : [$versions];
+        list($vsql, $vparams) = $DB->get_in_or_equal($versions, SQL_PARAMS_NAMED, 'ver');
+        $userfields = get_all_user_name_fields(true, 'u', null, 'user');
+        $userfieldsmod = get_all_user_name_fields(true, 'm', null, 'mod');
+        $sql = "SELECT u.id AS mainuserid, $userfields, $userfieldsmod, a.*, u.policyagreed
```

Comment:
`a.*` should be discouraged, explicit list of fields (even if all of them are enlisted) is better and safer, in case of future DB scheme changes / extensions (such as adding a field that would suddenly start to collide).